src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,204 @@
+        String aOrAn = firstCommand.equals("event") ? "an" : "a";
+        if (taskDesc.equals("")) {
+            throw new DukeException("☹ OOPS!!! The description of " + aOrAn + " " + firstCommand + " cannot be empty.");
+        } else if (date.equals("") && convertToTaskType(firstCommand) != Task.TaskType.TODO) {
+            throw new DukeException("☹ OOPS!!! The date of " + aOrAn + " " + firstCommand + " cannot be empty.");
+        } else if (convertToTaskType(firstCommand) == Task.TaskType.DEADLINE
+                || convertToTaskType(firstCommand) == Task.TaskType.EVENT) {
+            if (DATE_PATTERN.matcher(date).matches()) {
+
+                String[] dateSplit = date.split(" ");
+                String dateString = dateSplit[0];
+                String timeString = dateSplit[1];
+                LocalDate ld = LocalDate.parse(dateString);
+
+                this.tasks.addTask(taskDesc, convertToTaskType(firstCommand), ld, timeString);
+
+                this.confirmAdditionOfTask();
+            } else {
+                throw new DukeException("You need to put the date in yyyy-mm-dd hhmm format!");
+            }
+        } else {
+            this.tasks.addTask(taskDesc);
+            this.confirmAdditionOfTask();
+        }
+        this.writeDataToDuke();
+    }
+
+    /**
+     * Confirms the addition of a task.
+     */
+    public void confirmAdditionOfTask() {
+        int tasksLength = this.tasks.getTasksLength();
+        this.ui.showTaskAddedMessage(tasksLength, this.tasks.getTask(tasksLength).toString());
+    }
+
+    /**
+     * Method for duke.Duke to mark a task done.
+     * @throws DukeException Thrown when user gives an index out of range
+     * or not a number after the command done.
+     */
+    private void markDone() throws DukeException {
+        try {
+            int taskIndex = parser.findCommandIndex();
+            this.tasks.markTaskDone(taskIndex);
+            Task task = this.tasks.getTask(taskIndex);
+            this.ui.markTaskDone(task);
+            this.writeDataToDuke();
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("☹ OOPS!!! The number you gave is out of range!");
+        } catch (NumberFormatException e) {
+            throw new DukeException("☹ OOPS!!! Put a number after 'done'!");
+        }
+    }
+
+    /**
+     * Method used by Duke to convert task commands
+     * into the TaskType enum.
+     *
+     * @param command The command to be converted.
+     * @return The TaskType enum of the task.
+     */
+    public static Task.TaskType convertToTaskType(String command) {
+        if (command.equals("todo")) {
+            return Task.TaskType.TODO;
+        } else if (command.equals("event")) {
+            return Task.TaskType.EVENT;
+        } else {
+            return Task.TaskType.DEADLINE;
+        }
+    }
+
+    /**
+     * Main method for Duke.
+     * @param args Optional arguments for CLI.
```

Comment:
```suggestion
     * Main method for Duke.
     *
     * @param args Optional arguments for CLI.
```

I noticed this in other javadoc header comments as well! There should be an empty line between description and parameter section, as per the [Coding Standard](https://se-education.org/guides/conventions/java/intermediate.html).